test/e2e/persistent_volumes.go

```diff
@@ -114,14 +126,54 @@ func deletePVCandValidatePV(c *client.Client, ns string, pvc *api.PersistentVolu
 	return pv, pvc, nil
 }
 
+// Takes a map of PV:PVC bound pairs. Wraps deletePVCandValidatePV by calling the function in a loop over the supplied map.
+// Each call deletes a PVC and validates the paired PV for phase "Available".
+// Returns a newly constructed map.
+func deletePVCandValidatePVGroup(c *client.Client, ns string, pvPvcBindMap bindmap) (bindmap, error) {
+
+	var (
+		err                       error
+		tmpPv                     *api.PersistentVolume
+		retpvpvcpairs                 = make(bindmap)
+		pairCountIn, pairCountOut int = 0, 0
+	)
+
+	for i := 0; i < len(pvPvcBindMap); i++ {
+		pairCountIn++
+	}
+
+	for pv, pvc := range pvPvcBindMap {
+
+		if pvc != nil {
+			framework.Logf("Deleting PVC %v", pvc.Name)
+			tmpPv, _, err = deletePVCandValidatePV(c, ns, pvc, pv)
+			if err != nil {
+				return nil, err
```

Comment:
There's a general problem of not cleaning up (ie, not deleting) pvs and pvc when the bindMap is returned as nil. Eg. what if you've deleted half of the pvcs then get a delete error? The Context's AfterEach will see the nil and skip it's delete of all of the bound pvs/pvcs. So the bindMap needs to be returned, but which version of the map? The passed-in arg? The partially created new map? A merge of both?